storage/cloud-client/bucket_lock.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+from google.cloud import storage
+
+
+def set_retention_policy(bucket_name, retention_period):
+    """Defines a retention policy on a given bucket"""
+    # [START storage_set_retention_policy]
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+
+    bucket.retention_period = retention_period
+    print('Bucket {} retention period set for {} seconds'.format(
+        bucket.name,
+        bucket.retention_period))
+    # [END storage_set_retention_policy]
+
+
+def remove_retention_policy(bucket_name):
+    """Defines a retention policy on a given bucket"""
+    # [START storage_remove_retention_policy]
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+
+    bucket.retention_period = None
+    print('Bucket {} retention period set for {} seconds'.format(
+        bucket.name,
+        bucket.retention_period))
+    # [END storage_remove_retention_policy]
+
+
+def lock_retention_policy(bucket_name):
+    # [START storage_lock_retention_policy]
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+
+    # Warning: Once a retention policy is locked it cannot be unlocked
+    # and retention period can only be increased.
+    bucket.retention_policy_locked = True
+
+    print('Retention policy for {} is now locked'.format(bucket_name))
+    print('Retention policy Effective as of {}'.format(
+        bucket.retention_policy_effective_time))
```

Comment:
I get this error for this line. Any ideas? AttributeError: 'Bucket' object has no attribute 'retention_policy_effective_time'

Reply:
To lock the retention policy a patch operation needs to be performed. Review tests in google-cloud-python [PR#5534](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5534/) and w.r.t to this sample please review [system.py line#1268](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5534/files#diff-76380c4d34ce398e5075ab86d5305f5dR1268).

Reply:
(I updated my comment -- made a mistake 😄)